### DISCUSSION TITLE
ErrorException: Undefined array key 0

### DISCUSSION THREAD

**Discussion by micraux** [awcodes/filament-table-repeater#103] [Q&A] [answered]
Hello,

I can't figure why I am getting an `Undefined array key 0` error...

Everything is fine when I use the native Repeater field.

If I remove the `TextInput`, an empty TableRepeater appears, showing the correct number of lines: so I guess the relationship works as expected.

Any idea?

**Comment by awcodes** [comment]
v3 introduced breaking changes, you will have to update your headers() method. Please see the upgrade notes on the README.md

**Reply by micraux** [reply]
Sorry, I didn't mention that I was not upgrading an existing installation.
I am using the latest Filament version (3.2.25) and I just installed your plugin, following the steps in the documentation.
By the way, I am not using the headers() method: I am just trying to implement the Repeater/TableRepeater with only one TextInput.

**Reply by awcodes** [reply, marked as answer]
You still have to provide headers then hide them with `->renderHeader(false)`. It's to keep the table accessible. 

If it's only one field though maybe the simple repeater is all you need. https://filamentphp.com/docs/3.x/forms/fields/repeater#simple-repeaters-with-one-field

**Reply by micraux** [reply]
Ok, that works!
I was aware of the simple repeater, but I have multiple fields.

Thank you!

**Comment by DeviJv** [comment]
i have some problem after update to latest version 
im using array push to fill my repeater loop 
how to fix it bro?

**Reply by micraux** [reply]
I guess if you have the same problem, the selected answer should help you...

Have you provided the `header()` method?

**Reply by DeviJv** [reply]
solved bro 
thx u
